### PR TITLE
Bugfix warn_outputdt_release_desync for NaN release times

### PR DESCRIPTION
**Pull request by erikvansebille**
Fixing bug when the `_warn_outputdt_release_desync()` warning is raised if particle release times are NaN

### CODE REVIEW

File: parcels/particleset.py [hunk 1]
```diff
@@ -1078,7 +1078,7 @@ def execute(
         outputdt = output_file.outputdt if output_file else np.inf
         if isinstance(outputdt, timedelta):
             outputdt = outputdt.total_seconds()
-        if outputdt is not None:
+        if np.isfinite(outputdt) is not None:
```
Comment (VeckoTheGecko) [inline parcels/particleset.py:1081]:
```suggestion
        if np.isfinite(outputdt):
```

File: parcels/particleset.py [hunk 2]
```diff
@@ -1235,7 +1235,7 @@ def execute(
 
 def _warn_outputdt_release_desync(outputdt: float, release_times: Iterable[float]):
     """Gives the user a warning if the release time isn't a multiple of outputdt."""
-    if any(t % outputdt != 0 for t in release_times):
+    if any((np.isfinite(t) and t % outputdt != 0) for t in release_times):
```
Comment (VeckoTheGecko) [inline parcels/particleset.py:1238]: Is this change required? If `t = np.nan` then `t % outputdt != 0` would always be False anyway.

**Review by VeckoTheGecko — CHANGES_REQUESTED** [review]

File: parcels/particleset.py (hunk 2 above)
Comment (erikvansebille) [reply parcels/particleset.py:1238]: Well, without this extra statement the Parcels tutorial gives these warnings whenever the time is not set
Screenshot 2024-10-28 at 08 37 01

Comment (VeckoTheGecko) [reply parcels/particleset.py:1238]: 👍  :)

**Review by VeckoTheGecko — APPROVED** [review]

**Review by VeckoTheGecko — APPROVED** [review]
